test/fuse_pointwise.cpp

```diff
@@ -361,4 +361,43 @@ TEST_CASE(no_input)
     EXPECT(p == p2);
 }
 
+TEST_CASE(add_reshape_add)
+{
+    migraphx::shape s1{migraphx::shape::float_type, {3, 10, 16}};
+    migraphx::shape s2{migraphx::shape::float_type, {3, 40, 2, 2}};
+    migraphx::shape s3{migraphx::shape::float_type, {3, 10, 4, 2, 2}};
+    migraphx::program p1;
+    {
+        auto* mm  = p1.get_main_module();
+        auto x    = mm->add_parameter("x", s1);
+        auto y    = mm->add_parameter("y", s1);
+        auto z    = mm->add_parameter("z", s2);
+        auto add1 = mm->add_instruction(migraphx::make_op("add"), x, y);
+        auto reshape =
+            mm->add_instruction(migraphx::make_op("reshape", {{"dims", s2.lens()}}), add1);
```

Comment:
Add test with non-standard squeeze+contiguous.

Reply:
Squeeze doesnt require contiguous.

Reply:
Yeah, I mean Matcher would look for `squeeze, unsqueeze, reshape` or `flatten` op and skip match `contiguous` if there is any during the matching. I meant a test case like that. 

`Reshape` requires a standard shape right now.  

I am not sure if any pointwise would generate a non-standard shape but if we can have tests like the following : 
`pointwise ---> (non-standard output) --> contiguous --> "reshaper" op --> pointwise`. 

and see if it gets matched and handled well by the logic. 

In this case, inputs to the first pointwise could be non-standard shaped and the second pointwise should be able to handle non-standard shape as well if contiguous is not present. 

For example : 

`pointwise -->  (non standard output) --> squeeze/unsqueeze --> pointwise. 

Squeeze/unsqueeze can handle non-standard shapes.

Reply:
> Yeah, I mean Matcher would look for squeeze, unsqueeze, reshape or flatten op and skip match contiguous if there is any during the matching. I meant a test case like that.

Test case like what?

> Reshape requires a standard shape right now.

No it doesn't, but it doesnt work for all layouts, which is why we insert a contiguous. eliminate_contiguous can figure out if the contiguous is needed.

Reply:
> Test case like what?

`pointwise --> (non standard output) --> squeeze/unsqueeze --> pointwise.